src/main/java/com/amazon/opendistroforelasticsearch/security/auth/BackendRegistry.java

```diff
@@ -494,7 +500,9 @@ public boolean authenticate(final RestRequest request, final RestChannel channel
 
         if(authenticated) {
             final User impersonatedUser = impersonate(request, authenticatedUser);
-            threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, impersonatedUser==null?authenticatedUser:impersonatedUser);
+            final User loggedInUser = impersonatedUser==null?authenticatedUser:impersonatedUser;
+            threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, loggedInUser);
+            threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER_STRING, loggedInUser.getUserString());
```

Comment:
Why do we need to populate both?

Reply:
This is just to get the user details out to the plugins apart from security (by directly not taking dependency on security plugin). Set only for the REST requests after auth.

Reply:
How is it related to user injection for transport requests?

Reply:
This gets the user details and plugins can use this to inject via transport.

Reply:
Please split into 2 PRs. This is not a requirement for user injection into transport.

Reply:
Done. Removed this part in the current PR